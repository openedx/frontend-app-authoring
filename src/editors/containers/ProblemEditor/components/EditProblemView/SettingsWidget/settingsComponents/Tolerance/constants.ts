import messages from './messages';

export const ToleranceTypes = {
  percent: {
    type: 'Percent',
    message: messages.typesPercentage,
  },
  number: {
    type: 'Number',
    message: messages.typesNumber,
  },
  none: {
    type: 'None',
    message: messages.typesNone,
  },
} as const;
