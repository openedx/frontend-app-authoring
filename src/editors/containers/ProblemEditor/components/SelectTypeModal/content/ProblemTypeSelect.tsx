import React from 'react';
import { Button, Container } from '@openedx/paragon';
import { FormattedMessage } from '@edx/frontend-platform/i18n';

// SelectableBox in paragon has a bug where you can't change selection. So we override it
import SelectableBox from '../../../../../sharedComponents/SelectableBox';
import {
  ProblemTypes,
  ProblemTypeKeys,
  AdvanceProblemKeys,
  AdvancedProblemType,
  ProblemType,
} from '../../../../../data/constants/problem';
import messages from './messages';

interface Props {
  selected: ProblemType;
  setSelected: (selected: ProblemType | AdvancedProblemType) => void;
}

const ProblemTypeSelect: React.FC<Props> = ({
  selected,
  setSelected,
}) => {
  const handleChange = e => setSelected(e.target.value);
  const handleClick = () => setSelected(AdvanceProblemKeys.BLANK);
  const settings = { type: 'radio' };

  return (
    <Container style={{ width: '494px', height: '400px' }}>
      <SelectableBox.Set
        name="problem-type"
        columns={1}
        onChange={handleChange}
        type={settings.type}
        value={selected}
      >
        {Object.values(ProblemTypeKeys).map((key) => (
          key !== 'advanced'
            ? (
              <SelectableBox
                className="border border-light-400 text-primary-500 shadow-none"
                id={key}
                key={key}
                value={key}
                {...settings}
              >
                {ProblemTypes[key].title}
              </SelectableBox>
            )
            : null
        ))}
      </SelectableBox.Set>
      <Button variant="link" className="pl-0 mt-2" onClick={handleClick}>
        <FormattedMessage {...messages.advanceProblemButtonLabel} />
      </Button>
    </Container>
  );
};

export default ProblemTypeSelect;
