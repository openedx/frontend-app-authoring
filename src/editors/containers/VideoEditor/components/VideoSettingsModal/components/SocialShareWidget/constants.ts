export const analyticsEvents = {
  socialSharingSettingChanged: 'edx.social.video_sharing_setting.changed',
};
