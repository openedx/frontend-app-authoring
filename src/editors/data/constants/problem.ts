import { StrictDict } from '../../utils';
import singleSelect from '../images/singleSelect.png';
import multiSelect from '../images/multiSelect.png';
import dropdown from '../images/dropdown.png';
import numericalInput from '../images/numericalInput.png';
import textInput from '../images/textInput.png';
import advancedOlxTemplates from './advancedOlxTemplates';
import basicOlxTemplates from './basicOlxTemplates';

export const ProblemTypeKeys = StrictDict({
  SINGLESELECT: 'multiplechoiceresponse',
  MULTISELECT: 'choiceresponse',
  DROPDOWN: 'optionresponse',
  NUMERIC: 'numericalresponse',
  TEXTINPUT: 'stringresponse',
  ADVANCED: 'advanced',
} as const);
export type ProblemType = typeof ProblemTypeKeys[keyof typeof ProblemTypeKeys];

export const ProblemTypes = StrictDict({
  [ProblemTypeKeys.SINGLESELECT]: {
    title: 'Single select',
    preview: singleSelect,
    previewDescription: 'Learners must select the correct answer from a list of possible options.',
    description: 'Enter your single select answers below and select which choices are correct. Learners must choose one correct answer.',
    helpLink: 'https://edx.readthedocs.io/projects/edx-partner-course-staff/en/latest/exercises_tools/multiple_choice.html',
    prev: ProblemTypeKeys.TEXTINPUT,
    next: ProblemTypeKeys.MULTISELECT,
    template: basicOlxTemplates.singleSelect,

  },
  [ProblemTypeKeys.MULTISELECT]: {
    title: 'Multi-select',
    preview: multiSelect,
    previewDescription: 'Learners must select all correct answers from a list of possible options.',
    description: 'Enter your multi select answers below and select which choices are correct. Learners must choose all correct answers.',
    helpLink: 'https://edx.readthedocs.io/projects/edx-partner-course-staff/en/latest/exercises_tools/checkbox.html',
    next: ProblemTypeKeys.DROPDOWN,
    prev: ProblemTypeKeys.SINGLESELECT,
    template: basicOlxTemplates.multiSelect,
  },
  [ProblemTypeKeys.DROPDOWN]: {
    title: 'Dropdown',
    preview: dropdown,
    previewDescription: 'Learners must select the correct answer from a list of possible options',
    description: 'Enter your dropdown answers below and select which choice is correct. Learners must select one correct answer.',
    helpLink: 'https://edx.readthedocs.io/projects/edx-partner-course-staff/en/latest/exercises_tools/dropdown.html',
    next: ProblemTypeKeys.NUMERIC,
    prev: ProblemTypeKeys.MULTISELECT,
    template: basicOlxTemplates.dropdown,
  },
  [ProblemTypeKeys.NUMERIC]: {
    title: 'Numerical input',
    preview: numericalInput,
    previewDescription: 'Specify one or more correct numeric answers, submitted in a response field.',
    description: 'Enter correct numerical input answers below. Learners must enter one correct answer.',
    helpLink: 'https://edx.readthedocs.io/projects/edx-partner-course-staff/en/latest/exercises_tools/numerical_input.html',
    next: ProblemTypeKeys.TEXTINPUT,
    prev: ProblemTypeKeys.DROPDOWN,
    template: basicOlxTemplates.numeric,
  },
  [ProblemTypeKeys.TEXTINPUT]: {
    title: 'Text input',
    preview: textInput,
    previewDescription: 'Specify one or more correct text answers, including numbers and special characters, submitted in a response field.',
    description: 'Enter your text input answers below and select which choices are correct. Learners must enter one correct answer.',
    helpLink: 'https://edx.readthedocs.io/projects/edx-partner-course-staff/en/latest/exercises_tools/text_input.html',
    prev: ProblemTypeKeys.NUMERIC,
    next: ProblemTypeKeys.SINGLESELECT,
    template: basicOlxTemplates.textInput,
  },
  [ProblemTypeKeys.ADVANCED]: {
    title: 'Advanced Problem',
    preview: ('<div />'),
    description: 'An Advanced Problem Type',
    helpLink: 'something.com',
  },
});

export const AdvanceProblemKeys = StrictDict({
  BLANK: 'blankadvanced',
  CIRCUITSCHEMATIC: 'circuitschematic',
  JSINPUT: 'jsinputresponse',
  CUSTOMGRADER: 'customgrader',
  IMAGE: 'imageresponse',
  FORMULA: 'formularesponse',
  PROBLEMWITHHINT: 'problemwithhint',
} as const);
export type AdvancedProblemType = typeof AdvanceProblemKeys[keyof typeof AdvanceProblemKeys];

export function isAdvancedProblemType(pt: ProblemType | AdvancedProblemType): pt is AdvancedProblemType {
  return Object.values(AdvanceProblemKeys).includes(pt as any);
}

export const AdvanceProblems = StrictDict({
  [AdvanceProblemKeys.BLANK]: {
    title: 'Blank problem',
    status: '',
    template: '<problem></problem>',
  },
  [AdvanceProblemKeys.CIRCUITSCHEMATIC]: {
    title: 'Circuit schematic builder',
    status: 'Not supported',
    template: advancedOlxTemplates.circuitSchematic,
  },
  [AdvanceProblemKeys.JSINPUT]: {
    title: 'Custom JavaScript display and grading',
    status: '',
    template: advancedOlxTemplates.jsInputResponse,
  },
  [AdvanceProblemKeys.CUSTOMGRADER]: {
    title: 'Custom Python-evaluated input',
    status: 'Provisional',
    template: advancedOlxTemplates.customGrader,
  },
  [AdvanceProblemKeys.IMAGE]: {
    title: 'Image mapped input',
    status: 'Not supported',
    template: advancedOlxTemplates.imageResponse,
  },
  [AdvanceProblemKeys.FORMULA]: {
    title: 'Math expression input',
    status: '',
    template: advancedOlxTemplates.formulaResponse,
  },
  [AdvanceProblemKeys.PROBLEMWITHHINT]: {
    title: 'Problem with adaptive hint',
    status: 'Not supported',
    template: advancedOlxTemplates.problemWithHint,
  },
} as const);

export const ShowAnswerTypesKeys = StrictDict({
  ALWAYS: 'always',
  ANSWERED: 'answered',
  ATTEMPTED: 'attempted',
  CLOSED: 'closed',
  FINISHED: 'finished',
  CORRECT_OR_PAST_DUE: 'correct_or_past_due',
  PAST_DUE: 'past_due',
  NEVER: 'never',
  AFTER_SOME_NUMBER_OF_ATTEMPTS: 'after_attempts',
  AFTER_ALL_ATTEMPTS: 'after_all_attempts',
  AFTER_ALL_ATTEMPTS_OR_CORRECT: 'after_all_attempts_or_correct',
  ATTEMPTED_NO_PAST_DUE: 'attempted_no_past_due',
} as const);

export const ShowAnswerTypes = StrictDict({
  [ShowAnswerTypesKeys.ALWAYS]: {
    id: 'authoring.problemeditor.settings.showanswertype.always',
    defaultMessage: 'Always',
  },
  [ShowAnswerTypesKeys.ANSWERED]: {
    id: 'authoring.problemeditor.settings.showanswertype.answered',
    defaultMessage: 'Answered',
  },
  [ShowAnswerTypesKeys.ATTEMPTED]: {
    id: 'authoring.problemeditor.settings.showanswertype.attempted',
    defaultMessage: 'Attempted or Past Due',
  },
  [ShowAnswerTypesKeys.CLOSED]: {
    id: 'authoring.problemeditor.settings.showanswertype.closed',
    defaultMessage: 'Closed',
  },
  [ShowAnswerTypesKeys.FINISHED]: {
    id: 'authoring.problemeditor.settings.showanswertype.finished',
    defaultMessage: 'Finished',
  },
  [ShowAnswerTypesKeys.CORRECT_OR_PAST_DUE]: {
    id: 'authoring.problemeditor.settings.showanswertype.correct_or_past_due',
    defaultMessage: 'Correct or Past Due',
  },
  [ShowAnswerTypesKeys.PAST_DUE]: {
    id: 'authoring.problemeditor.settings.showanswertype.past_due',
    defaultMessage: 'Past Due',
  },
  [ShowAnswerTypesKeys.NEVER]: {
    id: 'authoring.problemeditor.settings.showanswertype.never',
    defaultMessage: 'Never',
  },
  [ShowAnswerTypesKeys.AFTER_SOME_NUMBER_OF_ATTEMPTS]: {
    id: 'authoring.problemeditor.settings.showanswertype.after_attempts',
    defaultMessage: 'After Some Number of Attempts',
  },
  [ShowAnswerTypesKeys.AFTER_ALL_ATTEMPTS]: {
    id: 'authoring.problemeditor.settings.showanswertype.after_all_attempts',
    defaultMessage: 'After All Attempts',
  },
  [ShowAnswerTypesKeys.AFTER_ALL_ATTEMPTS_OR_CORRECT]: {
    id: 'authoring.problemeditor.settings.showanswertype.after_all_attempts_or_correct',
    defaultMessage: 'After All Attempts or Correct',
  },
  [ShowAnswerTypesKeys.ATTEMPTED_NO_PAST_DUE]: {
    id: 'authoring.problemeditor.settings.showanswertype.attempted_no_past_due',
    defaultMessage: 'Attempted',
  },
} as const);

export const RandomizationTypesKeys = StrictDict({
  NEVER: 'never',
  ALWAYS: 'always',
  ONRESET: 'onreset',
  PERSTUDENT: 'per_student',
} as const);

export const RandomizationTypes = StrictDict({
  [RandomizationTypesKeys.ALWAYS]: {
    id: 'authoring.problemeditor.settings.RandomizationTypes.always',
    defaultMessage: 'Always',
  },
  [RandomizationTypesKeys.NEVER]: {
    id: 'authoring.problemeditor.settings.RandomizationTypes.never',
    defaultMessage: 'Never',
  },
  [RandomizationTypesKeys.ONRESET]: {
    id: 'authoring.problemeditor.settings.RandomizationTypes.onreset',
    defaultMessage: 'On Reset',
  },
  [RandomizationTypesKeys.PERSTUDENT]: {
    id: 'authoring.problemeditor.settings.RandomizationTypes.perstudent',
    defaultMessage: 'Per Student',
  },
} as const);

export const RichTextProblems = [ProblemTypeKeys.SINGLESELECT, ProblemTypeKeys.MULTISELECT] as const;

export const settingsOlxAttributes = [
  '@_display_name',
  '@_weight',
  '@_max_attempts',
  '@_showanswer',
  '@_show_reset_button',
  '@_submission_wait_seconds',
  '@_attempts_before_showanswer_button',
] as const;
