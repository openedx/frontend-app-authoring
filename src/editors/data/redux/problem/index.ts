export { actions, reducer } from './reducers';
export { default as selectors } from './selectors';
