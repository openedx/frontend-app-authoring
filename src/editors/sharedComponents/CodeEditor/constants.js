// HTML symbols to unescape in a Alphanumeric value : Literal mapping.
const alphanumericMap = {
  cent: '¢',
  pound: '£',
  sect: '§',
  copy: '©',
  laquo: '«',
  raquo: '»',
  reg: '®',
  deg: '°',
  plusmn: '±',
  para: '¶',
  middot: '·',
  frac12: '½',
  mdash: '—',
  ndash: '–',
  lsquo: '‘',
  rsquo: '’',
  sbquo: '‚',
  rdquo: '”',
  ldquo: '“',
  dagger: '†',
  Dagger: '‡',
  bull: '•',
  hellip: '…',
  prime: '′',
  Prime: '″',
  euro: '€',
  trade: '™',
  asymp: '≈',
  ne: '≠',
  le: '≤',
  ge: '≥',
  quot: '"',
};
export default alphanumericMap;

export const HEADING = '## Heading\n\n';
export const MULTIPLE_CHOICE = '( ) Option 1\n( ) Option 2\n(x) Correct\n';
export const CHECKBOXES = '[ ] Incorrect\n[x] Correct\n';
export const TEXT_INPUT = 'Type your answer here: ___\n';
export const NUMERICAL_INPUT = '= 100 +-5\n';
export const DROPDOWN = '[Dropdown: Option A\nOption B\nCorrect Option* ]\n';
export const EXPLANATION = '>> Add explanation text here <<\n';
