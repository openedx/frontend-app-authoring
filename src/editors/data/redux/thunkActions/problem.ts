import _ from 'lodash';
import { actions as problemActions } from '../problem';
import { actions as requestActions } from '../requests';
import { selectors as appSelectors } from '../app';
import * as requests from './requests';
import { isLibraryKey } from '../../../../generic/key-utils';
import { OLXParser } from '../../../containers/ProblemEditor/data/OLXParser';
import { parseSettings } from '../../../containers/ProblemEditor/data/SettingsParser';
import { ProblemTypeKeys } from '../../constants/problem';
import ReactStateOLXParser from '../../../containers/ProblemEditor/data/ReactStateOLXParser';
import { camelizeKeys } from '../../../utils';
import { fetchEditorContent } from '../../../containers/ProblemEditor/components/EditProblemView/hooks';
import { RequestKeys } from '../../constants/requests';

// Similar to `import { actions, selectors } from '..';` but avoid circular imports:
const actions = { problem: problemActions, requests: requestActions };
const selectors = { app: appSelectors };

export const switchToAdvancedEditor = () => (dispatch, getState) => {
  const state = getState();
  const editorObject = fetchEditorContent({ format: '' });
  const reactOLXParser = new ReactStateOLXParser({ problem: state.problem, editorObject });
  const rawOLX = reactOLXParser.buildOLX();
  dispatch(actions.problem.updateField({ problemType: ProblemTypeKeys.ADVANCED, rawOLX }));
};

export const isBlankProblem = ({ rawOLX }) => {
  if (['<problem></problem>', '<problem/>'].includes(rawOLX.replace(/\s/g, ''))) {
    return true;
  }
  return false;
};

export const getDataFromOlx = ({ rawOLX, rawSettings, defaultSettings }) => {
  let olxParser;
  let parsedProblem;
  const { default_to_advanced: defaultToAdvanced } = rawSettings;
  try {
    olxParser = new OLXParser(rawOLX);
    if (defaultToAdvanced) {
      parsedProblem = olxParser.getBetaParsedOLXData();
    } else {
      parsedProblem = olxParser.getParsedOLXData();
    }
  } catch (error) {
    // eslint-disable-next-line no-console
    console.error('The Problem Could Not Be Parsed from OLX. redirecting to Advanced editor.', error);
    return { problemType: ProblemTypeKeys.ADVANCED, rawOLX, settings: parseSettings(rawSettings, defaultSettings) };
  }
  if (parsedProblem?.problemType === ProblemTypeKeys.ADVANCED) {
    return { problemType: ProblemTypeKeys.ADVANCED, rawOLX, settings: parseSettings(rawSettings, defaultSettings) };
  }
  const { settings, ...data } = parsedProblem;
  const parsedSettings = { ...settings, ...parseSettings(rawSettings, defaultSettings) };
  if (!_.isEmpty(rawOLX) && !_.isEmpty(data)) {
    return { ...data, rawOLX, settings: parsedSettings };
  }
  return { settings: parsedSettings };
};

export const loadProblem = ({ rawOLX, rawSettings, defaultSettings }) => (dispatch) => {
  if (isBlankProblem({ rawOLX })) {
    dispatch(actions.problem.setEnableTypeSelection(camelizeKeys(defaultSettings)));
  } else {
    dispatch(actions.problem.load(getDataFromOlx({ rawOLX, rawSettings, defaultSettings })));
  }
};

export const fetchAdvancedSettings = ({ rawOLX, rawSettings }) => (dispatch) => {
  const advancedProblemSettingKeys = ['max_attempts', 'showanswer', 'show_reset_button', 'rerandomize'];
  dispatch(requests.fetchAdvancedSettings({
    onSuccess: (response) => {
      const defaultSettings = {};
      Object.entries(response.data as Record<string, any>).forEach(([key, value]) => {
        if (advancedProblemSettingKeys.includes(key)) {
          defaultSettings[key] = value.value;
        }
      });
      dispatch(actions.problem.updateField({ defaultSettings: camelizeKeys(defaultSettings) }));
      loadProblem({ rawOLX, rawSettings, defaultSettings })(dispatch);
    },
    onFailure: () => { loadProblem({ rawOLX, rawSettings, defaultSettings: {} })(dispatch); },
  }));
};

export const initializeProblem = (blockValue) => (dispatch, getState) => {
  const rawOLX = _.get(blockValue, 'data.data', {});
  const rawSettings = _.get(blockValue, 'data.metadata', {});
  const learningContextId = selectors.app.learningContextId(getState());
  if (isLibraryKey(learningContextId)) {
    // Content libraries don't yet support defaults for fields like max_attempts, showanswer, etc.
    // So proceed with loading the problem.
    // Though first we need to fake the request or else the problem type selection UI won't display:
    dispatch(actions.requests.completeRequest({ requestKey: RequestKeys.fetchAdvancedSettings, response: {} }));
    dispatch(loadProblem({ rawOLX, rawSettings, defaultSettings: {} }));
  } else {
    // Load the defaults (for max_attempts, etc.) from the course's advanced settings, then proceed:
    dispatch(fetchAdvancedSettings({ rawOLX, rawSettings }));
  }
};

export default { initializeProblem, switchToAdvancedEditor, fetchAdvancedSettings };
