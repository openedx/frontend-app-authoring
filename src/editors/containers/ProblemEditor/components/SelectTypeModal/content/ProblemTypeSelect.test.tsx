import 'CourseAuthoring/editors/setupEditorTest';

import { shallow } from '@edx/react-unit-test-utils';
import { ProblemTypeKeys } from '../../../../../data/constants/problem';
import ProblemTypeSelect from './ProblemTypeSelect';

describe('ProblemTypeSelect', () => {
  const props = {
    setSelected: jest.fn(),
  };

  describe('snapshot', () => {
    test('SINGLESELECT', () => {
      expect(shallow(
        <ProblemTypeSelect {...props} selected={ProblemTypeKeys.SINGLESELECT} />,
      ).snapshot).toMatchSnapshot();
    });
    test('MULTISELECT', () => {
      expect(shallow(
        <ProblemTypeSelect {...props} selected={ProblemTypeKeys.MULTISELECT} />,
      ).snapshot).toMatchSnapshot();
    });
    test('DROPDOWN', () => {
      expect(shallow(
        <ProblemTypeSelect {...props} selected={ProblemTypeKeys.DROPDOWN} />,
      ).snapshot).toMatchSnapshot();
    });
    test('NUMERIC', () => {
      expect(shallow(
        <ProblemTypeSelect {...props} selected={ProblemTypeKeys.NUMERIC} />,
      ).snapshot).toMatchSnapshot();
    });
    test('TEXTINPUT', () => {
      expect(shallow(
        <ProblemTypeSelect {...props} selected={ProblemTypeKeys.TEXTINPUT} />,
      ).snapshot).toMatchSnapshot();
    });
  });
});
