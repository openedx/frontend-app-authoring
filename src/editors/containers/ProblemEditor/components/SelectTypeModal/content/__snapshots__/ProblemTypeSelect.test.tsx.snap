// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProblemTypeSelect snapshot DROPDOWN 1`] = `
<Container
  style={
    {
      "height": "400px",
      "width": "494px",
    }
  }
>
  <ForwardRef
    columns={1}
    name="problem-type"
    onChange={[Function]}
    type="radio"
    value="optionresponse"
  >
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="multiplechoiceresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="multiplechoiceresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="multiplechoiceresponse"
    >
      Single select
    </ForwardRef>
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="choiceresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="choiceresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="choiceresponse"
    >
      Multi-select
    </ForwardRef>
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="optionresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="optionresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="optionresponse"
    >
      Dropdown
    </ForwardRef>
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="numericalresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="numericalresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="numericalresponse"
    >
      Numerical input
    </ForwardRef>
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="stringresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="stringresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="stringresponse"
    >
      Text input
    </ForwardRef>
  </ForwardRef>
  <Button
    className="pl-0 mt-2"
    onClick={[Function]}
    variant="link"
  >
    <FormattedMessage
      defaultMessage="Advanced problem types"
      description="Button label for advance problem types option"
      id="authoring.problemEditor.problemSelect.advanceButton.label"
    />
  </Button>
</Container>
`;

exports[`ProblemTypeSelect snapshot MULTISELECT 1`] = `
<Container
  style={
    {
      "height": "400px",
      "width": "494px",
    }
  }
>
  <ForwardRef
    columns={1}
    name="problem-type"
    onChange={[Function]}
    type="radio"
    value="choiceresponse"
  >
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="multiplechoiceresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="multiplechoiceresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="multiplechoiceresponse"
    >
      Single select
    </ForwardRef>
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="choiceresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="choiceresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="choiceresponse"
    >
      Multi-select
    </ForwardRef>
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="optionresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="optionresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="optionresponse"
    >
      Dropdown
    </ForwardRef>
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="numericalresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="numericalresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="numericalresponse"
    >
      Numerical input
    </ForwardRef>
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="stringresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="stringresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="stringresponse"
    >
      Text input
    </ForwardRef>
  </ForwardRef>
  <Button
    className="pl-0 mt-2"
    onClick={[Function]}
    variant="link"
  >
    <FormattedMessage
      defaultMessage="Advanced problem types"
      description="Button label for advance problem types option"
      id="authoring.problemEditor.problemSelect.advanceButton.label"
    />
  </Button>
</Container>
`;

exports[`ProblemTypeSelect snapshot NUMERIC 1`] = `
<Container
  style={
    {
      "height": "400px",
      "width": "494px",
    }
  }
>
  <ForwardRef
    columns={1}
    name="problem-type"
    onChange={[Function]}
    type="radio"
    value="numericalresponse"
  >
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="multiplechoiceresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="multiplechoiceresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="multiplechoiceresponse"
    >
      Single select
    </ForwardRef>
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="choiceresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="choiceresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="choiceresponse"
    >
      Multi-select
    </ForwardRef>
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="optionresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="optionresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="optionresponse"
    >
      Dropdown
    </ForwardRef>
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="numericalresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="numericalresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="numericalresponse"
    >
      Numerical input
    </ForwardRef>
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="stringresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="stringresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="stringresponse"
    >
      Text input
    </ForwardRef>
  </ForwardRef>
  <Button
    className="pl-0 mt-2"
    onClick={[Function]}
    variant="link"
  >
    <FormattedMessage
      defaultMessage="Advanced problem types"
      description="Button label for advance problem types option"
      id="authoring.problemEditor.problemSelect.advanceButton.label"
    />
  </Button>
</Container>
`;

exports[`ProblemTypeSelect snapshot SINGLESELECT 1`] = `
<Container
  style={
    {
      "height": "400px",
      "width": "494px",
    }
  }
>
  <ForwardRef
    columns={1}
    name="problem-type"
    onChange={[Function]}
    type="radio"
    value="multiplechoiceresponse"
  >
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="multiplechoiceresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="multiplechoiceresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="multiplechoiceresponse"
    >
      Single select
    </ForwardRef>
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="choiceresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="choiceresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="choiceresponse"
    >
      Multi-select
    </ForwardRef>
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="optionresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="optionresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="optionresponse"
    >
      Dropdown
    </ForwardRef>
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="numericalresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="numericalresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="numericalresponse"
    >
      Numerical input
    </ForwardRef>
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="stringresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="stringresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="stringresponse"
    >
      Text input
    </ForwardRef>
  </ForwardRef>
  <Button
    className="pl-0 mt-2"
    onClick={[Function]}
    variant="link"
  >
    <FormattedMessage
      defaultMessage="Advanced problem types"
      description="Button label for advance problem types option"
      id="authoring.problemEditor.problemSelect.advanceButton.label"
    />
  </Button>
</Container>
`;

exports[`ProblemTypeSelect snapshot TEXTINPUT 1`] = `
<Container
  style={
    {
      "height": "400px",
      "width": "494px",
    }
  }
>
  <ForwardRef
    columns={1}
    name="problem-type"
    onChange={[Function]}
    type="radio"
    value="stringresponse"
  >
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="multiplechoiceresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="multiplechoiceresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="multiplechoiceresponse"
    >
      Single select
    </ForwardRef>
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="choiceresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="choiceresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="choiceresponse"
    >
      Multi-select
    </ForwardRef>
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="optionresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="optionresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="optionresponse"
    >
      Dropdown
    </ForwardRef>
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="numericalresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="numericalresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="numericalresponse"
    >
      Numerical input
    </ForwardRef>
    <ForwardRef
      checked={false}
      className="border border-light-400 text-primary-500 shadow-none"
      id="stringresponse"
      inputHidden={true}
      isIndeterminate={false}
      isInvalid={false}
      key="stringresponse"
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value="stringresponse"
    >
      Text input
    </ForwardRef>
  </ForwardRef>
  <Button
    className="pl-0 mt-2"
    onClick={[Function]}
    variant="link"
  >
    <FormattedMessage
      defaultMessage="Advanced problem types"
      description="Button label for advance problem types option"
      id="authoring.problemEditor.problemSelect.advanceButton.label"
    />
  </Button>
</Container>
`;
